### PR TITLE
Add htmlTitle prop to Icon

### PR DESCRIPTION
**Pull request by gabeboning**
#### Checklist

- [x] [Enable CircleCI for your fork](https://circleci.com/add-projects)
- [x] [Enable preview comments on CircleCI](https://github.com/palantir/blueprint/blob/develop/CONTRIBUTING.md#enable-preview-comments)
- [ ] Include tests
- [ ] Update documentation

#### Changes proposed in this pull request:

Add an `htmlTitle` prop to `<Icon>` which gets applied as the HTML `title` attribute, and appears as the built in browser tooltip on hover

Helps address/provide workaround for #3214

### CODE REVIEW

**Comment by gabeboning** [comment]
Comment preview bot just logged them as output:

documentation: https://circleci.com/gh/palantir/blueprint/22908/artifacts/0//home/circleci/project/packages/docs-app/dist/index.html
landing: https://circleci.com/gh/palantir/blueprint/22908/artifacts/0//home/circleci/project/packages/landing-app/dist/index.html
table: https://circleci.com/gh/palantir/blueprint/22908/artifacts/0//home/circleci/project/packages/table-dev-app/dist/index.html

File: packages/core/src/components/icon/icon.tsx [hunk 1]
```diff
@@ -65,6 +65,12 @@ export interface IIconProps extends IIntentProps, IProps {
      * explicit falsy value to disable.
      */
     title?: string | false | null;
+
+    /**
+     * String for the title attribute on the rendered element, which will appear
```
Comment (giladgray) [inline packages/core/src/components/icon/icon.tsx]: backticks around `title`

File: packages/core/src/components/icon/icon.tsx [hunk 2]
```diff
@@ -65,6 +65,12 @@ export interface IIconProps extends IIntentProps, IProps {
...
+     * as a tooltip in the browser.
```
Comment (giladgray) [inline packages/core/src/components/icon/icon.tsx]: "as a native browser tooltip."

File: packages/core/src/components/icon/icon.tsx [hunk 3]
```diff
@@ -88,6 +94,7 @@ export class Icon extends React.PureComponent<IIconProps & React.DOMAttributes<H
             intent,
             title = icon,
             tagName: TagName = "span",
+            htmlTitle,
```
Comment (giladgray) [inline packages/core/src/components/icon/icon.tsx]: nit: alphabetize

**Pull request merged**